pandas_gbq/gbq.py

```diff
@@ -837,68 +681,172 @@ def _get_credentials_file():
         'PANDAS_GBQ_CREDENTIALS_FILE')
 
 
-def _parse_data(schema, rows):
-    # see:
-    # http://pandas.pydata.org/pandas-docs/dev/missing_data.html
-    # #missing-data-casting-rules-and-indexing
-    dtype_map = {'FLOAT': np.dtype(float),
-                 'TIMESTAMP': 'M8[ns]'}
+def sizeof_fmt(num, suffix='B'):
+    fmt = "%3.1f %s%s"
+    for unit in ['', 'K', 'M', 'G', 'T', 'P', 'E', 'Z']:
+        if abs(num) < 1024.0:
+            return fmt % (num, unit, suffix)
+        num /= 1024.0
+    return fmt % (num, 'Y', suffix)
 
-    fields = schema['fields']
-    col_types = [field['type'] for field in fields]
-    col_names = [str(field['name']) for field in fields]
-    col_dtypes = [dtype_map.get(field['type'], object) for field in fields]
-    page_array = np.zeros((len(rows),), dtype=lzip(col_names, col_dtypes))
-    for row_num, raw_row in enumerate(rows):
-        entries = raw_row.get('f', [])
-        for col_num, field_type in enumerate(col_types):
-            field_value = _parse_entry(entries[col_num].get('v', ''),
-                                       field_type)
-            page_array[row_num][col_num] = field_value
 
-    return DataFrame(page_array, columns=col_names)
+def run_query(query, client, dialect='legacy', query_parameters=(),
+              configuration=None, verbose=True, async=True):
+    """Execute a query job
 
+    Parameters
+    ----------
+    query, dialect, query_paramaters, configuration, verbose : see read_gbq()
+    client : bigQuery Client object
+        Client with the specified project_id and credentials used to run the
+        query
+    async: bool
+        Whether a synchronous or asynchronous query should be run. To be
+        deprecated in future versions; synchronous queries are used as a
+        workaround to implement timeouts, and will be removed in a
+        future update once Google Cloud Python resolves the issue.
 
-def _parse_entry(field_value, field_type):
-    if field_value is None or field_value == 'null':
-        return None
-    if field_type == 'INTEGER':
-        return int(field_value)
-    elif field_type == 'FLOAT':
-        return float(field_value)
-    elif field_type == 'TIMESTAMP':
-        timestamp = datetime.utcfromtimestamp(float(field_value))
-        return np.datetime64(timestamp)
-    elif field_type == 'BOOLEAN':
-        return field_value == 'true'
-    return field_value
+    Returns
+    -------
+    Tuple
+        rows : list of lists
+        columns: list of strings
+        schema: dictionary
+            Has the following keys: name, field_type, mode, fields, description
+    """
+
+    def _wait_for_job(job):
```

Comment:
This function is unnecessary as of [google-cloud-bigquery version 0.26.0](https://github.com/GoogleCloudPlatform/google-cloud-python/releases/tag/bigquery-0.26.0), which implements the `concurrent.futures` API for all job types.

Reply:
Removed